mlflow/system_metrics/__init__.py

```diff
@@ -0,0 +1,47 @@
+"""System metrics logging module."""
+
+from mlflow.environment_variables import (
+    MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING,
+    MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING,
+    MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL,
+)
+from mlflow.utils.annotations import experimental
+
+
+@experimental
+def disable_system_metrics_logging():
+    """Disable system metrics logging globally.
+
+    Calling this function will disable system metrics logging globally, but users can still opt in
+    system metrics logging for individual runs by `mlflow.start_run(log_system_metrics=True)`.
+    """
+    MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING.set(False)
+
+
+@experimental
+def enable_system_metrics_logging():
+    """Enable system metrics logging globally.
+
+    Calling this function will enable system metrics logging globally, but users can still opt out
+    system metrics logging for individual runs by `mlflow.start_run(log_system_metrics=False)`.
+    """
+    MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING.set(True)
+
+
+@experimental
+def set_system_metrics_sampling_interval(interval):
+    """Set the system metrics sampling interval.
+
+    Every `interval` seconds, the system metrics will be collected. By default `interval=10`.
+    """
+    MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL.set(interval)
+
+
+@experimental
+def set_system_metrics_samples_before_logging(samples):
```

Comment:
```suggestion
def set_system_metrics_sampling_size(samples):
```

`sampling_size`?

Reply:
hmm is `_size` a bit vague?

Reply:
Just a suggestion, `set_system_metrics_samples_before_logging` sounds ok.

Reply:
thanks Haru!